### PR TITLE
[Merged by Bors] - public metrics

### PR DESCRIPTION
**Pull request by dshulyak**
subset of metrics to record:
- connections
- version
- init size started / completed
- post proof time

they will be recorded with 5 first characters from p2p public key, we probably don't want to gather information about smesher key.

example config:
```json
{
    "main": {
        "metrics-push": "https://public-metrics-gateway.spacemesh.dev/",
        "metrics-push-period": "1s",
        "metrics-push-user": "XXX",
        "metrics-push-pass": "XXXXXX",
        "metrics-push-header": {
            "X-Scope-OrgId": "XXXXXXX"
        }
    }
}
```

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -1374,9 +1375,13 @@ func (app *App) Start(ctx context.Context) error {
...
-		metrics.StartPushingMetrics(app.Config.MetricsPush,
-			app.Config.MetricsPushUser, app.Config.MetricsPushPass, app.Config.MetricsPushPeriod,
-			app.host.ID().String(), app.Config.Genesis.GenesisID().ShortString())
+		metrics.StartPushingMetrics(
+			app.Config.MetricsPush,
+			app.Config.MetricsPushUser,
+			app.Config.MetricsPushPass,
+			app.Config.MetricsPushHeader,
+			app.Config.MetricsPushPeriod,
+			app.host.ID().String()[:5], app.Config.Genesis.GenesisID().ShortString())
```
Comment (countvonzero) [inline node/node.go:1384]: what do you think about hashing the first 5 char?

**Review by countvonzero — APPROVED** [review]

File: node/node.go (hunk 1 above)
Comment (dshulyak) [reply node/node.go:1384]: maybe hash all then and use 5 chars? i can do

Comment (dshulyak) [reply node/node.go:1384]: actually it doesn't hide anything, if i want i can hash all identities found in the network and match them

Comment (countvonzero) [reply node/node.go:1384]: yes. but at least its a step towards anomynizing. somebody have to pay the computation to get it....

Comment (pigmej) [reply node/node.go:1384]: But the same info will be publicly available in the epoch when ATX will be published, right? So why are we trying to hide it?

Comment (countvonzero) [reply node/node.go:1384]: this is p2p id

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**